Signals/Domain/MissingValuePolicy/FirstOrderMissingValuePolicy.cs

```diff
@@ -40,24 +49,30 @@ public class FirstOrderMissingValuePolicy<T> : MissingValuePolicy<T>
 
         private Quality GetMinQuality(Quality a, Quality b)
         {
-            // TODO should we add numbers to Quality? Those would probably go into Dto and suggest some impl...
             if (a == Quality.None || b == Quality.None)
                 return Quality.None;
-            return (Quality)Math.Max((int)a, (int)b);
+
+            return Enum
+                .GetValues(typeof(Quality))
+                .Cast<Quality>()
+                .Last(q => q == a || q == b);
         }
 
         private Datum<T> Interpolate(Datum<T> older, Datum<T> newer, DateTime currentTs)
         {
-            var q = GetMinQuality(older.Quality, newer.Quality);
-            if (q == Quality.None)
+            var resultQuality = GetMinQuality(older.Quality, newer.Quality);
+
+            if (resultQuality == Quality.None)
                 return Datum<T>.CreateNone(Signal, currentTs);
+
             var timeSpan = (newer.Timestamp - older.Timestamp).TotalSeconds;
-            var valueSpan = (dynamic)newer.Value - (dynamic)older.Value; // yuck!
```

Comment:
for a moment I was considering `Convert` to `decimal` but it's probably even yuckier 

if that code will ever made 'production' probably some optimizations will be required